irohad/ametsuchi/impl/redis_block_query.cpp

```diff
@@ -141,35 +150,96 @@ namespace iroha {
           });
     }
 
-    rxcpp::observable<model::Transaction>
-    RedisBlockQuery::getAccountAssetTransactions(const std::string &account_id,
-                                                 const std::string &asset_id) {
-      return rxcpp::observable<>::create<model::Transaction>(
-          [this, account_id, asset_id](auto subscriber) {
-            auto block_ids = this->getBlockIds(account_id);
-            if (block_ids.empty()) {
-              subscriber.on_completed();
-              return;
-            }
+    bool RedisBlockQuery::hasAccountAssetRelatedCommand(
+        const std::string &account_id,
+        const std::vector<std::string> &assets_id,
+        const std::shared_ptr<iroha::model::Command> &command) const {
+      return isCommandValid<model::TransferAsset>(
+        command,
+        [&account_id, &assets_id](const auto &transfer) {
+          return (transfer.src_account_id == account_id
+                  or transfer.dest_account_id == account_id)
```

Comment:
This two lambdas are same except this line. You can introduce generic one and chose the proper one with the std::bind